### PR TITLE
repo: Remove more packages to free diskspace in Docker job

### PR DESCRIPTION
**Pull request by phlax**
Signed-off-by: Ryan Northey <ryan@synca.io>

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by phlax** [comment]
cc @mattklein123

**Comment by phlax** [comment]
this frees up a couple of extra GB which will most likely resolve the failing docker ci in postsubmit due to disk space

if it does work, it will probably only kick the can down the road

not sure how we can resolve better - i reckon we can remove more packages, as done here, with a bit more testing

im also wondering about whether the job can be restructured to be more like build -> push -> delete for the images, so it frees up space as each image is created - not sure about this

it might also be worth at least checking why the images have increased in size, altho, without debugging this it looks as tho it has just grown slowly to use up the remaining space

there have been a couple of other disk-space issue in the presubmit queue - unrelated to this PR, other than they may be also be a consequence of increased disk usage

**Review by adisuissa — APPROVED** [review]
LGTM, thanks!

File: .azure-pipelines/cleanup.sh [hunk 1]
```diff
@@ -2,24 +2,37 @@
...
+    "temurin-*-jdk"
+    "zulu-*-azure-jdk")
 
 sudo apt-get update -y || true
 sudo apt-get purge -y --no-upgrade "${PURGE_PACKAGES[@]}"
 
 dpkg-query -Wf '${Installed-Size}\t${Package}\n' | sort -rn
+
+echo "Disk space after package removal"
+df -h
```
Comment (adisuissa) [inline .azure-pipelines/cleanup.sh:38]: I'm assuming this (and the above) are needed for debug, right?

**Pull request merged**